projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java

```diff
@@ -29,26 +31,80 @@
     extends RecyclerView.Adapter<SpamRecyclerAdapter.RecyclerViewHolder> {
   private static final String TAG = "SpamRecyclerAdapter";
 
+  ArrayList<SmsMessage> inboxSpam;
+
   public SpamRecyclerAdapter() {
     Handler handler = new Handler(Looper.myLooper());
+    inboxSpam = new ArrayList<>();
+
+    // Fetch spams from DB that are in user's inbox
+    for (int i = 0; i < SmsMessages.get().getInboxMessages().size(); i++) {
+      for (SmsMessage sms : SmsMessages.get().getDbMessages()) {
+        if (SmsMessages.get().getInboxMessages().get(i).getBody().contentEquals(sms.getBody())
+            && SmsMessages.get()
+                .getInboxMessages()
+                .get(i)
+                .getSender()
+                .contentEquals(sms.getSender())) {
+          // Add message from the DB - keep received date of user's message
+          inboxSpam.add(
+              sms.toBuilder()
+                  .setReceivedAt(SmsMessages.get().getInboxMessages().get(i).getReceivedAt())
+                  .build());
+        }
+      }
+    }
 
-    // Listen to db smsMessages
+    // Listen to db smsMessages. Update just relevant messages that are in the inbox as well.
+    // Keep the received date of the user's message.
     SmsMessages.get()
         .setMessagesListener(
             new SmsMessages.MessagesListener() {
               @Override
               public void messageAdded(SmsMessage newMessage) {
-                handler.post(() -> notifyItemInserted(SmsMessages.get().getDbMessages().size()));
+                int inboxIndex =
+                    SmsMessages.get()
+                        .searchMessage(newMessage, SmsMessages.get().getInboxMessages());
+                if (inboxIndex != -1) {
+                  inboxSpam.add(
+                      newMessage
+                          .toBuilder()
+                          .setReceivedAt(
+                              SmsMessages.get().getInboxMessages().get(inboxIndex).getReceivedAt())
+                          .build());
+                  handler.post(() -> notifyItemInserted(inboxSpam.size()));
+                }
               }
 
               @Override
               public void messageModified(int index) {
-                handler.post(() -> notifyItemChanged(index));
+                int inboxSpamIndex =
+                    SmsMessages.get()
+                        .searchMessage(SmsMessages.get().getDbMessages().get(index), inboxSpam);
+                if (inboxSpamIndex != -1) {
```

Comment:
What are you doing here? having hard time following

Reply:
It looks a little "messy" to me too, that's why I added the comment above all of these functions. (Add / Modify / Remove)
>// Listen to db smsMessages. Update just relevant messages that are in the inbox as well.
>// Keep the received date of the user's message.

I'll add another comment for this function specifically...

Reply:
maybe with the cached list it will be easier to read.
Plus the way it written
as 
a 
single
thing
every
line
kinda make it hard to read. I think ff doing it and it is annoying...